ArduPlane/takeoff.cpp

```diff
@@ -179,66 +180,111 @@ void Plane::takeoff_calc_roll(void)
  */
 void Plane::takeoff_calc_pitch(void)
 {
-    if (auto_state.highest_airspeed < g.takeoff_rotate_speed) {
-        // we have not reached rotate speed, use the specified takeoff target pitch angle
-        nav_pitch_cd = int32_t(100.0f * mode_takeoff.ground_pitch);
-        return;
+    // First see if TKOFF_ROTATE_SPD applies.
+    // This will set the pitch for the first portion of the takeoff, up until cruise speed is reached.
+    if (g.takeoff_rotate_speed > 0) {
+        // A non-zero rotate speed is recommended for ground takeoffs.
+        if (auto_state.highest_airspeed < g.takeoff_rotate_speed) {
+            // We have not reached rotate speed, use the specified takeoff target pitch angle.
+            nav_pitch_cd = int32_t(100.0f * mode_takeoff.ground_pitch);
+            TECS_controller.set_pitch_min(0.01f*nav_pitch_cd);
+            TECS_controller.set_pitch_max(0.01f*nav_pitch_cd);
+            return;
+        } else if (gps.ground_speed() <= (float)aparm.airspeed_cruise) {
+            // If rotate speed applied, gradually transition from TKOFF_GND_PITCH to the climb angle.
+            // This is recommended for ground takeoffs, so delay rotation until ground speed indicates adequate airspeed.
+            const uint16_t min_pitch_cd = 500; // Set a minimum of 5 deg climb angle.
+            nav_pitch_cd = (gps.ground_speed() / (float)aparm.airspeed_cruise) * auto_state.takeoff_pitch_cd;
+            nav_pitch_cd = constrain_int32(nav_pitch_cd, min_pitch_cd, auto_state.takeoff_pitch_cd); 
+            TECS_controller.set_pitch_min(0.01f*nav_pitch_cd);
```

Comment:
this doesn't look right, this is using fixed pitch after rotate and before cruise speed. That doesn't sound right at all, we won't be able to rotate

Reply:
we've lost the case of with an airspeed sensor and doing rotating rolling takeoff